cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/AnnotateDiagnostics.java

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package io.ballerina.cli.utils;
+
+import io.ballerina.compiler.internal.diagnostics.StringDiagnosticProperty;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.internal.PackageDiagnostic;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import io.ballerina.tools.diagnostics.Location;
+import io.ballerina.tools.text.TextDocument;
+
+import java.util.ArrayList;
+
+import static io.ballerina.cli.utils.DiagnosticAnnotation.SEVERITY_COLORS;
+
+/**
+ * This class is used to generate diagnostic annotations from diagnostics.
+ *
+ * @since 2201.9.0
+ */
+public class AnnotateDiagnostics {
+
+    public static String renderDiagnostic(Diagnostic diagnostic, Document document, int terminalWidth) {
+
+        String diagnosticCode = diagnostic.diagnosticInfo().code();
+        if (diagnosticCode.startsWith("BCE")) {
+            int diagnosticCodeNumber = Integer.parseInt(diagnosticCode.substring(3));
+            if (diagnosticCodeNumber < 1000) {
+                PackageDiagnostic packageDiagnostic = (PackageDiagnostic) diagnostic;
+                return diagnosticToString(diagnostic) + "\n" + getSyntaxDiagnosticAnnotation(
+                        document, packageDiagnostic, diagnosticCodeNumber, terminalWidth);
```

Comment:
For `\n`  can we use the, `System.getProperty("line.separator")`

Reply:
Better if we can use `System.lineSeparator()` instead